### PR TITLE
[opencypher compatibility] Support using exists with dynamic map

### PR DESCRIPTION
**Pull request by Aiee**
As title.
Support query: `match (n:player) WHERE exists(n['name']) return n`
Fixes https://github.com/vesoft-inc/nebula-graph/issues/841

Depends on https://github.com/vesoft-inc/nebula-common/pull/509

### CODE REVIEW

File: tests/tck/features/expression/Predicate.feature [hunk 1]
```diff
@@ -49,6 +49,26 @@ Feature: Predicate
       | "Manu Ginobili"     | 95       |
       | "Tim Duncan"        | 95       |
 
+  Scenario: exists with dynamic map in MATCH
+    Given a graph with space named "nba"
+    When executing query:
+      """
+      MATCH(n:player) WHERE EXISTS(n['name'])
```
Comment (Shylock-Hg) [inline tests/tck/features/expression/Predicate.feature:56]: Add a test case for only a part vertex contains property.

Comment (Aiee) [reply tests/tck/features/expression/Predicate.feature:74]: ACK.

Comment (nevermore3) [reply tests/tck/features/expression/Predicate.feature:74]: please add case : match (a:player )-[b:like|:serve]->(c) where exists(b['likeness']) return a

Comment (Aiee) [reply tests/tck/features/expression/Predicate.feature:73]: > please add case : match (a:player )-[b:like|:serve]->(c) where exists(b['likeness']) return a

This case has been added. Thanks for pointing that out.

Comment (czpmango) [reply tests/tck/features/expression/Predicate.feature:73]: `match (v:player)-[]->(n) return exists(n["age"])`

**Review by nevermore3 — APPROVED** [review]

File: tests/tck/features/expression/Predicate.feature [hunk 2]
```diff
@@ -49,6 +49,99 @@ Feature: Predicate
       | "Manu Ginobili"     | 95       |
       | "Tim Duncan"        | 95       |
 
+  Scenario: exists with dynamic map in MATCH
+    Given a graph with space named "nba"
```
Comment (yixinglu) [inline tests/tck/features/expression/Predicate.feature:53]: Use a new space to test following cases and drop it at last.

**Review by yixinglu — COMMENTED** [review]
Test variable length pattern match.

**Review by yixinglu — DISMISSED** [review]

**Review by jievince — APPROVED** [review]

File: tests/tck/features/expression/Predicate.feature [hunk 3]
```diff
@@ -49,6 +49,118 @@ Feature: Predicate
...
+      """
+    Then a SyntaxError should be raised at runtime: The exists only accept LabelAttribe, Attribute and Subscript
+    Then drop the used space
+
+  Scenario: exists with dynamic map in variable length MATCH
+    Given a graph with space named "nba"
+    When executing query:
+      """
+      MATCH(v:player{name:"Tim Duncan"})-[e:like|serve*2]->(v2)
+      RETURN DISTINCT e, ALL(e IN e WHERE EXISTS(e['likeness']))
```
Comment (yixinglu) [inline tests/tck/features/expression/Predicate.feature:152]: Is is valid for this same variable name `e` as list variable name `e` @jievince ?

Comment (jievince) [reply tests/tck/features/expression/Predicate.feature:152]: yes, it's valid. Because after the parser, the inner var has been rewrited to another name

**Review by jievince — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Pull request merged**